### PR TITLE
[JAVA-4584] Enabled core-java-12 module

### PR DESCRIPTION
**Pull request by panos-kakos**

### CODE REVIEW

File: core-java-modules/core-java-12/pom.xml [hunk 1]
```diff
@@ -30,21 +30,18 @@
                 <configuration>
```
Comment (anastasiosioannidis) [inline core-java-modules/core-java-12/pom.xml:30]: try removing the maven-compiler-plugin

File: core-java-modules/core-java-13/pom.xml [hunk 2]
```diff
@@ -22,24 +22,21 @@
                 <configuration>
                     <source>${maven.compiler.source.version}</source>
```
Comment (anastasiosioannidis) [inline core-java-modules/core-java-13/pom.xml:23]: try removing the maven-compiler-plugin

File: core-java-modules/core-java-13/pom.xml [hunk 3]
```diff
@@ -22,24 +22,21 @@
...
-                </configuration>
             </plugin>
         </plugins>
     </build>
 
+
     <properties>
-        <maven.compiler.source.version>13</maven.compiler.source.version>
-        <maven.compiler.target.version>13</maven.compiler.target.version>
+        <java.version>17</java.version>
```
Comment (anastasiosioannidis) [inline core-java-modules/core-java-13/pom.xml:37]: Remove the maven compiler properties and leave only java.version

File: core-java-modules/core-java-15/pom.xml [hunk 4]
```diff
@@ -34,25 +33,25 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>${maven-compiler-plugin.version}</version>
                 <configuration>
-                    <release>${maven.compiler.release}</release>
-                    <compilerArgs>--enable-preview</compilerArgs>
-                    <source>14</source>
-                    <target>14</target>
```
Comment (anastasiosioannidis) [inline core-java-modules/core-java-15/pom.xml:40]: try removing the maven-compiler-plugin

File: core-java-modules/core-java-15/pom.xml [hunk 5]
```diff
@@ -34,25 +33,25 @@
...
-                <configuration>
-                    <argLine>--enable-preview</argLine>
-                </configuration>
             </plugin>
         </plugins>
     </build>
 
     <properties>
-        <maven.compiler.release>15</maven.compiler.release>
+        <java.version>17</java.version>
```
Comment (anastasiosioannidis) [inline core-java-modules/core-java-15/pom.xml:52]: Remove the maven compiler properties and leave only java.version

**Comment by panos-kakos** [comment]
test this please

File: core-java-modules/core-java-15/pom.xml (hunk 5 above)
Comment (panos-kakos) [reply core-java-modules/core-java-15/pom.xml:30]: done

File: core-java-modules/core-java-15/pom.xml (hunk 4 above)
Comment (panos-kakos) [reply core-java-modules/core-java-15/pom.xml:40]: done

File: core-java-modules/core-java-13/pom.xml (hunk 3 above)
Comment (panos-kakos) [reply core-java-modules/core-java-13/pom.xml:17]: done

File: core-java-modules/core-java-13/pom.xml (hunk 2 above)
Comment (panos-kakos) [reply core-java-modules/core-java-13/pom.xml]: done

File: core-java-modules/core-java-12/pom.xml [hunk 6]
```diff
@@ -30,21 +30,18 @@
...
-                    <argLine>--enable-preview</argLine>
-                </configuration>
             </plugin>
         </plugins>
     </build>
 
     <properties>
-        <maven.compiler.source.version>12</maven.compiler.source.version>
-        <maven.compiler.target.version>12</maven.compiler.target.version>
+        <java.version>17</java.version>
```
[the remark this answers is not in the document]
Comment (panos-kakos) [reply core-java-modules/core-java-12/pom.xml:25]: done

File: core-java-modules/core-java-12/pom.xml (hunk 1 above)
Comment (panos-kakos) [reply core-java-modules/core-java-12/pom.xml]: done

**Pull request merged**